### ISSUE TITLE
Add way to distinguish success and success-with-errors for continue-on-error jobs

### ISSUE THREAD

**Issue by rnett** [actions/runner#1161]
**Describe the enhancement**
I have a few jobs that I don't want to fail the workflow on failure.  However, I do want to be able to react to the failure in further jobs, i.e. with `needs.A.result`.  This is currently impossible: `continue-on-error` means that even if the job errors, the result is set to `success` (note that the UI still shows failure).

I would like for either `result` to be set to `failed` or a new state (`errored`?) when a `continue-on-error` job errors, or for there to be some other way to detect errors in `continue-on-error` jobs.  The best solution is probably what is currently done for [steps](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context): a pre-`continue-on-error` output (e.g. `outcome`) and a post one (e.g. `conclusion`, but could keep the name `result`)

**Comment by ethomson** [comment]
Yeah, this makes sense @rnett - I appreciate the feedback.  I agree that this is a little clunky right now.  I'll make sure that this is on the backlog for us to give some thought to.
